lib/geokit/geocoders/ca_geocoder.rb

```diff
@@ -18,25 +18,24 @@ class CaGeocoder < Geocoder
 
       # Template method which does the geocode lookup.
       def self.do_geocode(loc)
-        raise ArgumentError("Geocoder.ca requires a GeoLoc argument") unless loc.is_a?(GeoLoc)
-        process :xml, submit_url(loc), loc
+        process :xml, submit_url(loc), GeoLoc.new
       end
 
       def self.parse_xml(xml, loc)
-        loc.lat = xml.elements["//latt"].text
-        loc.lng = xml.elements["//longt"].text
-        loc.success = true
+        loc.lat             = xml.elements['//latt'].text
+        loc.lng             = xml.elements['//longt'].text
+        loc.city            = xml.elements['//city'].text
+        loc.street_number   = xml.elements['//stnumber'].text
+        loc.street_address  = xml.elements['//staddress'].text
+        loc.province        = xml.elements['//prov'].text
```

Comment:
Unnecessary spacing detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.